Example/src/screens/Animations.tsx

```diff
@@ -0,0 +1,180 @@
+import {View, StyleSheet} from 'react-native';
+import {enableScreens} from 'react-native-screens';
+import {
+  createNativeStackNavigator,
+  NativeStackNavigationProp,
+} from 'react-native-screens/native-stack';
+import {SettingsPicker, Button} from '../shared';
+
+enableScreens();
+
+type StackParamList = {
+  Main: undefined;
+  Push: undefined;
+  Pop: undefined;
+  Screen: undefined;
+  Modal: undefined;
+};
+
+interface MainScreenProps {
+  navigation: NativeStackNavigationProp<StackParamList, 'Main'>;
+  stackAnimation: string;
+  setStackAnimation: (value: string) => void;
+}
+
+const MainScreen = ({
+  navigation,
+  stackAnimation,
+  setStackAnimation,
+}: MainScreenProps): JSX.Element => {
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      stackAnimation,
+    });
+  }, [navigation, stackAnimation]);
+
+  return (
+    <View style={styles.container}>
+      <SettingsPicker
+        label="Stack animation"
+        value={stackAnimation}
+        onValueChange={setStackAnimation}
+        items={[
+          'default',
+          'fade',
+          'flip',
+          'slide_from_right',
+          'slide_from_left',
+          'none',
+        ]}
+      />
+      <Button
+        title="Replace with pop animation"
+        onPress={() => navigation.replace('Pop')}
+      />
+      <Button
+        title="Replace with push animation"
+        onPress={() => navigation.replace('Push')}
+      />
+      <Button
+        title="New screen"
+        onPress={() => navigation.navigate('Screen')}
+      />
+      <Button title="Open modal" onPress={() => navigation.navigate('Modal')} />
+      <Button onPress={() => navigation.pop()} title="🔙 Back to Examples" />
+    </View>
+  );
+};
+
+interface ReplaceScreenProps {
+  navigation: NativeStackNavigationProp<StackParamList>;
+  stackAnimation: string;
+}
+
+const ReplaceScreen = ({
+  navigation,
+  stackAnimation,
+}: ReplaceScreenProps): JSX.Element => {
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      stackAnimation,
+    });
+  }, [navigation, stackAnimation]);
+
+  return (
+    <View style={styles.container}>
+      <Button title="Go back" onPress={() => navigation.replace('Main')} />
+    </View>
+  );
+};
+
+interface NavigateScreenProps {
+  navigation: NativeStackNavigationProp<StackParamList>;
+  stackAnimation: string;
+}
+
+const NavigateScreen = ({
+  navigation,
+  stackAnimation,
+}: NavigateScreenProps): JSX.Element => {
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      stackAnimation,
+    });
+  }, [navigation, stackAnimation]);
+
+  return (
+    <View style={styles.container}>
+      <Button title="Go back" onPress={() => navigation.navigate('Main')} />
+    </View>
+  );
+};
+
+const Stack = createNativeStackNavigator<StackParamList>();
+
+const App = (): JSX.Element => {
+  const [stackAnimation, setStackAnimation] = useState('default');
+  return (
+    <Stack.Navigator
+      screenOptions={{
+        statusBarStyle: 'dark',
```

Comment:
Why is this option here?

Reply:
Fixed in https://github.com/software-mansion/react-native-screens/pull/827/commits/85e8e17b756652507b71956a30319822422c5358